isolated-jobs-test/src/main/java/com/hazelcast/jet/tests/isolatedjobs/IsolatedJobsBatchTest.java

```diff
@@ -0,0 +1,140 @@
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.tests.common.Util.sleepMillis;
+import static com.hazelcast.jet.tests.isolatedjobs.JetMemberSelectorUtil.excludeMember;
+import static java.util.stream.Collectors.joining;
+
+/**
+ * This test requires Enterprise License with Advanced Compute Feature.
+ * To run this test locally, you have to use EE version (it can be forced by adding dependency to hazelcast-enterprise
+ * in the pom.xml) and license must be set (it can be done in {@link AbstractSoakTest#run(String[])})
+ */
+public class IsolatedJobsBatchTest extends AbstractSoakTest {
+    public static final int DEFAULT_SLEEP_BETWEEN_NEW_JOBS_MS = 5_000;
+    public static final String BATCH_SINK_PREFIX = "IsolatedJobsTestBatchSink";
+    private static final int LOG_JOB_COUNT_THRESHOLD = 60;
+    private int sleepBetweenNewJobsMs;
+
+    public static void main(final String[] args) throws Exception {
+        new IsolatedJobsBatchTest().run(args);
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        sleepBetweenNewJobsMs = propertyInt("sleepBetweenNewJobsMs",
+                DEFAULT_SLEEP_BETWEEN_NEW_JOBS_MS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        List<UUID> membersUuids = client.getCluster().getMembers().stream()
+                .map(Member::getUuid)
+                .collect(Collectors.toList());
+
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            UUID excludedMember = getRandomMemberUUID(membersUuids);
+            createBatchJob(client, excludedMember, jobCount)
+                    .join();
+            assertBatchSink(client, jobCount, membersUuids, excludedMember);
+            clearSink(client, jobCount);
+
+            jobCount++;
+            if (jobCount % LOG_JOB_COUNT_THRESHOLD == 0) {
+                logger.info("Job count: " + jobCount);
+            }
+            sleepMillis(sleepBetweenNewJobsMs);
+        }
+        logger.info("Final job count: " + jobCount);
+    }
+
+    private void clearSink(HazelcastInstance client, long jobCount) {
+        client.getList(BATCH_SINK_PREFIX + jobCount).clear();
+    }
+
+    private void assertBatchSink(HazelcastInstance client, long jobCount, List<UUID> membersUuids, UUID excludedMember) {
+        List<UUID> expectedUuids = membersUuids
+                .stream()
+                .filter(uuid -> !uuid.equals(excludedMember))
+                .collect(Collectors.toList());
+
+        IList<UUID> list = client.getList(BATCH_SINK_PREFIX + jobCount);
+        //Check if job was not processed on excluded member
+        list.forEach(uuid -> {
```

Comment:
why we rather don't check that `list` doesn't contain `excludedMember` for assumption "Check if job was not processed on excluded member"?

Reply:
I made in this way to have better logging in case of issue.